deployments/charts/kuma/templates/cni-daemonset.yaml

```diff
@@ -51,14 +51,31 @@ spec:
       containers:
         - name: install-cni
           {{- if .Values.experimental.cni }}
-          image: {{ include "kuma.formatImage" (dict "image" .Values.cni.imageExperimental "root" $) | quote }}
+          image: {{ include "kuma.formatImage" (dict "image" .Values.cni.experimental.image "root" $) | quote }}
           imagePullPolicy: IfNotPresent
+          readinessProbe:
+            initialDelaySeconds: {{ .Values.cni.delayStartupSeconds }}
+            exec:
+              command:
+                - cat
+                - /tmp/ready
+          {{- if .Values.cni.delayStartupSeconds }}
+          command: [ "/bin/sh", "-c", "--" ]
+          args: [ "sleep {{.Values.cni.delayStartupSeconds}} && exec /install-cni" ]
+          {{- else }}
```

Comment:
What I meant in my previous comment. Is that if you rely on the default to be 0 and that a `sleep 0` is a noop. This can become simpler and not require a `if` in the helm which greatly improves readability

Reply:
I was worried a bit about spawning a new process but it will probably be alright, I'll remove the if statement